java/ql/lib/upgrades/initial/semmlecode.dbscheme

```diff
@@ -0,0 +1,648 @@
+@fieldAccessExpr = @readsFieldExpr | @writesFieldExpr;
+@callExpr = @makesMethodCallExpr | @makesConstructorCallExpr;
+
+@member = @method | @constructor | @field | @reftype ;
+
+@locatable = @file | @class | @interface | @fielddecl | @field | @constructor | @method | @param | @exception
+             | @boundedtype | @typebound | @array
+             | @readsFieldExpr | @writesFieldExpr | @makesMethodCallExpr | @makesConstructorCallExpr
+             | @import | @stmt | @expr | @localvar | @javadoc | @javadocTag | @javadocText 
+             | @xmllocatable | @externalDefect | @externalMetric;
+             
+@locatable_element = @file | @class | @interface | @field | @param | @exception | @callable
+
+@named = @param | @member | @package | @typeormethod | @file;
+
+@modifier_named = @modifier | @modifiable | @named;
+
+@javadocParent = @javadoc | @javadocTag;
+@javadocElement = @javadocTag | @javadocText;
+
+/* XML Files */ 
+
+xmlEncoding  (unique int id: @file ref, varchar(900) encoding: string ref);
+
+xmlDTDs       (unique int id: @xmldtd, 
+              varchar(900) root: string ref, 
+              varchar(900) publicId: string ref, 
+              varchar(900) systemId: string ref,
+              int fileid: @file ref); 
+             
+xmlElements   (unique int id: @xmlelement, 
+              varchar(900) name: string ref, 
+              int parentid: @xmlparent ref,
+              int idx: int ref,
+              int fileid: @file ref);            
+
+xmlAttrs      (unique int id: @xmlattribute, 
+              int elementid: @xmlelement ref,
+              varchar(900) name: string ref, 
+              varchar(3600) value: string ref,
+              int idx: int ref,
+              int fileid: @file ref);  
+               
+xmlNs         (int id: @xmlnamespace, 
+              varchar(900) prefixName: string ref, 
+              varchar(900) URI: string ref,
+              int fileid: @file ref);  
+               
+xmlHasNs      (int elementId: @xmlnamespaceable ref, 
+              int nsId: @xmlnamespace ref,
+              int fileid: @file ref);  
+
+xmlComments   (unique int id: @xmlcomment, 
+              varchar(3600) text: string ref, 
+              int parentid: @xmlparent ref,
+              int fileid: @file ref); 
+               
+xmlChars      (unique int id: @xmlcharacters, 
+              varchar(3600) text: string ref, 
+              int parentid: @xmlparent ref,
+              int idx: int ref,
+              int isCDATA: int ref,
+              int fileid: @file ref);  
+	           	
+@xmlparent = @file | @xmlelement;
+@xmlnamespaceable = @xmlelement | @xmlattribute;
+
+xmllocations(int xmlElement: @xmllocatable ref,
+			int location: @location_default ref);
+
+@xmllocatable = @xmlcharacters | @xmlelement | @xmlcomment | @xmlattribute | @xmldtd | @file | @xmlnamespace;
+
+@top = @element | @locatable | @folder;
+
+/* *** Well-Founded Orders *** */
+
+/* The type hierarchy order */
+order:typeOrder(3);

```

Comment:
It looks like this is causing an error for the QL extractor:
> [build-stdout] ERROR /home/runner/work/codeql/codeql/java/ql/lib/upgrades/initial/semmlecode.dbscheme:615: parse error

See https://github.com/github/codeql/runs/5160181021?check_suite_focus=true#step:8:37
But I am not sure if that is relevant.

Reply:
Ah, this is not relevant to the present PR, but it _is_ very relevant to the QL extractor. Thanks for bringing this to our attention!